sqlglot/parser.py

```diff
@@ -5475,6 +5484,44 @@ def _parse_comprehension(
             condition=condition,
         )
 
+    def _parse_heredoc(self) -> t.Optional[exp.Heredoc]:
+        tags = []
+        tag_text = None
+
+        if self._match_text_seq("$"):
+            tags.append("$")
+        else:
+            return None
+
+        if self._curr and self._is_connected():
+            if not self._match_text_seq("$"):
+                tag_text = self._curr.text
+                tags.append(tag_text.upper())
+                self._advance()
+            else:
+                tags.append("$")
```

Comment:
Perhaps you can replace this whole if-else block with `tags.append(self._advance() and self._prev.text.upper())`?

Reply:
@tobymao wdyt?

Reply:
sure

Reply:
ah may need to actually `self._advance()` before appending `self._prev.text.upper()` @viethungle-vt1401, sorry for this (it seems like it's related to the mypy issue)

Reply:
I tried specifying `return None`for `_advance()`, but that still didn't satisfy the linter, so going with your approach!